testbed/stabilitytests/metric_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"testing"
+
+	"go.opentelemetry.io/collector/testbed/testbed"
+	scenarios "go.opentelemetry.io/collector/testbed/tests"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/testbed/datareceivers"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/testbed/datasenders"
+)
+
+func TestStabilityMetricsOpenCensus(t *testing.T) {
+	scenarios.Scenario10kItemsPerSecond(
+		t,
+		testbed.NewOCMetricDataSender(testbed.GetAvailablePort(t)),
+		testbed.NewOCDataReceiver(testbed.GetAvailablePort(t)),
+		testbed.ResourceSpec{
+			ExpectedMaxCPU:      45,
+			ExpectedMaxRAM:      86,
+			ResourceCheckPeriod: resourceCheckPeriod,
+		},
+		nil,
+	)
+}
+
+func TestStabilityMetricsCarbon(t *testing.T) {
+	scenarios.Scenario10kItemsPerSecond(
+		t,
+		datasenders.NewCarbonDataSender(testbed.GetAvailablePort(t)),
+		datareceivers.NewCarbonDataReceiver(testbed.GetAvailablePort(t)),
+		testbed.ResourceSpec{
+			ExpectedMaxCPU:      237,
+			ExpectedMaxRAM:      90,
+			ResourceCheckPeriod: resourceCheckPeriod,
+		},
+		nil,
+	)
+}
+
+func TestStabilityMetricsSignalFx(t *testing.T) {
+	scenarios.Scenario10kItemsPerSecond(
+		t,
+		datasenders.NewSFxMetricDataSender(testbed.GetAvailablePort(t)),
+		datareceivers.NewSFxMetricsDataReceiver(testbed.GetAvailablePort(t)),
+		testbed.ResourceSpec{
+			ExpectedMaxCPU:      83,
+			ExpectedMaxRAM:      91,
+			ResourceCheckPeriod: resourceCheckPeriod,
+		},
+		nil,
+	)
+}
```

Comment:
Do these need to be 3 separate functions or can be a matrix test like TestTrace10kSPS?

Reply:
Can we also have OTLP test?

Reply:
Yes, we need them as separate tests. It makes it much easier to split between circleCI runners